projects/_posts/2016-08-07-Mapping.md

```diff
@@ -0,0 +1,198 @@
+---
+enable_x3d_support: true
+author: [ "@bmegli" ] 
+programming_language: [ "C/C++/C#" ] 
+
+#youtube_video_id: G6uVg34VzHw # The video ID of the YouTube video to be displayed with this post
+
+project_homepage_url: "https://github.com/bmegli/ev3dev-mapping" 
+source_code_url: "https://github.com/bmegli/ev3dev-mapping" 
+#building_instructions_url: "" # how to build the model out of LEGO (*not* how to build the source code)
+
+excerpt: "A real-time 3D mapping project." # Yet another ev3dev project.
+---
+
+## Overview
+
+A 2D/3D mapping/scanning project consisting off:
+
+- robot side code 
+- cross-platform visualization 
+
+Robot side code is really a bunch of super-simple programs which I call modules.
+Each of those programs works perfectly well from the command line without the rest of the project.
+The sole exception is ev3control module which enables/disables/monitors other modules.
+The repository for robot side code is [ev3dev-mapping-modules].
+
+The cross-platform visualization is a [Unity] project that can plot the readings in real-time and/or control the robot.
+Like other projects written in Unity it works on more than 10 [platforms], in theory at least. 
+Unity has also a nice abstraction for input so robots can be controlled with keyboards/pads/mice/wheels/mobiles and so on.
+The repository for cross-platform visualization is [ev3dev-mapping-ui].
+
+I try to keep the budget of the project low, around 400$ (360€) including EV3 and motors.
+
+The meta-repository of the project is [ev3dev-mapping].
+
+## Examples
+
+The examples are X3D models. You can rotate, move and zoom with the mouse.
+
+{% include x3d-model.html caption="Example Phase II raw scan" source="\images\projects\2016-08-07-Mapping\csi-body.x3d" %}
+
+{% include x3d-model.html caption="Example Phase II result + MeshLab surface reconstruction" source="\images\projects\2016-08-07-Mapping\face-reconstruction.x3d" %}
+
+
+
+## Phase I
+
+Goals:
+
+1. Hardware selection
+-  CruizCore XG1300L gyroscope for low bias drift and great accuracy
+-  XV11 Lidar for unmatched... price 
+2. Drivers
+-  CruizCore XG1300L driver on top of wonderful @dlech sensor driver framework
+-  XV11 Lidar in userspace using information from [xv11hacking]
+3. Correct geometry
+-  nobody (on the internet) seems to treat XV11 [rotational geometry] correctly
+4. LIDAR standard deviation
+-  evaluation with respect to distance
+5. Experimental 2D mapping
+-  experimental code to learn and throw away
+
+The Phase I is finished with results:
+
+-   [Using the XV11 Lidar tutorial]
+-   [xv11lidar] C/C++ library
+-   [mi-xg1300l] CruizCore XG1300L driver 
+-   [ev3dev-mapping-abandoned] repository
+
+The Phase I robot in action:
+
+-   [How to interface XV11 LIDAR to EV3 using ev3dev]
+-   [EV3 Gyro vs CruizCore XG1300L vs Odometry - Position Estimation]
+
+{% include screenshot.html source="\images\projects\2016-08-07-Mapping\robot-phaseI.jpg" caption="Hardware used for testing in Phase I" %}
```

Comment:
This is not windows :wink:

`/`, not `\`

There are other places this needs to be fixed as well.